src/vocab.rs

```diff
@@ -34,6 +35,39 @@ impl PyVocab {
             }
         })
     }
+
+    fn ngram_indices(&self, word: &str) -> PyResult<Vec<Py<PyTuple>>> {
+        let embeds = self.embeddings.borrow();
+        let gil = pyo3::Python::acquire_gil();
+        match embeds.vocab() {
+            VocabWrap::FastTextSubwordVocab(inner) => Ok(inner
+                .ngram_indices(word)
+                .unwrap()
+                .into_iter()
+                .map(|pair| pair.into_py(gil.python()))
+                .collect()),
+            VocabWrap::FinalfusionSubwordVocab(inner) => Ok(inner
+                .ngram_indices(word)
+                .unwrap()
+                .into_iter()
+                .map(|pair| pair.into_py(gil.python()))
+                .collect()),
```

Comment:
I don't think this part looks good because of the repeat of code, so I also tried using `macro` to create syntax that allow matching `FinalfusionSubwordVocab(inner)` and `FastTextSubwordVocab(inner)` compactly. But I'm not so sure which way is better.

Reply:
After calling `ngram_indices` on both vocab types, you end up with the same type, `Option<Vec<(String, usize)>>`, so starting from that point you could return the result from those calls from the match and assign it to a variable.

If we stick to my suggestions from above, then the code-duplication would be eliminated anyways, since both match arms would consist of `Ok(inner.ngram_indices(word))`, I don't believe there is a more concise way to write that.